cmd/troubleshoot/cli/receive.go

```
@@ -0,0 +1,105 @@
+	if err != nil {
+		return err
+	}
+
+	bundlePath, err := ioutil.TempDir("", "troubleshoot")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(bundlePath)
+
+	receivedCollectors := []string{}
+	for {
+		job, err := troubleshootClient.CollectorJobs(collectorJobNamespace).Get(collectorJobName, metav1.GetOptions{})
+		if err != nil && kuberneteserrors.IsNotFound(err) {
+			// where did it go!
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		for _, readyCollector := range job.Status.Successful {
+			alreadyReceived := false
+			for _, receivedCollector := range receivedCollectors {
+				if receivedCollector == readyCollector {
+					alreadyReceived = true
+				}
+			}
+
+			if !alreadyReceived {
+				collectorResp, err := http.Get(fmt.Sprintf("http://localhost:8000/collector/%s", readyCollector))
+				if err != nil {
+					return err
+				}
+
+				defer collectorResp.Body.Close()
+				body, err := ioutil.ReadAll(collectorResp.Body)
+				if err != nil {
+					return err
+				}
+
+				decoded, err := base64.StdEncoding.DecodeString(string(body))
+				if err != nil {
+					return err
+				}
+
+				files := make(map[string][]byte)
+				if err := json.Unmarshal(decoded, &files); err != nil {
+					return err
+				}
+
+				for filename, contents := range files {
+					fileDir, fileName := filepath.Split(filename)
+					outPath := filepath.Join(bundlePath, fileDir)
+					if err := os.MkdirAll(outPath, 0777); err != nil {
+						return err
+					}
+
+					if err := ioutil.WriteFile(filepath.Join(outPath, fileName), contents, 0644); err != nil {
+						return err
+					}
+
+				}
+
+				receivedCollectors = append(receivedCollectors, readyCollector)
+			}
+		}
+
+		if len(job.Status.Running) == 0 {
+			tarGz := archiver.TarGz{
+				Tar: &archiver.Tar{
+					ImplicitTopLevelFolder: false,
+				},
+			}
+
+			paths := make([]string, 0, 0)
+			for _, id := range receivedCollectors {
+				paths = append(paths, filepath.Join(bundlePath, id))
```

if some errors are ignored, this should check if path exists